membership_extension/models/account_move_line.py

```diff
@@ -18,6 +18,8 @@ def create(self, vals_list):
         for line in lines:
             if line.move_id.move_type == "out_invoice" and line.product_id.membership:
                 line.membership_lines.write({"state": "waiting"})
+                for partner in line.membership_lines.mapped("partner"):
+                    partner.write({"free_member": False})
```

Comment:
suggestion: Same but simpler and faster:


```suggestion
                line.membership_lines.partner.free_member = False
```